lib/gist.rb

```diff
@@ -181,6 +184,55 @@ def list_gists(user = "")
     end
   end
 
+  def list_all_gists(user="")
+    url = "#{base_path}"
+
+    if user == ""
+      access_token = auth_token()
+      if access_token.to_s != ''
+        url << "/gists?access_token=" << CGI.escape(access_token)
+        get_gist_pages(url)
+      else
+        raise Error, "Not authenticated. Use 'gist --login' to login or 'gist -l username' to view public gists."
+      end
+
+    else
+      url << "/users/#{user}/gists"
+      get_gist_pages(url)
+    end
+
+  end
+
+  def get_gist_pages(url)
+
+    request = Net::HTTP::Get.new(url)
+    response = http(api_url, request)
+
+    m = response.header['link'].match(/page=(\d+)>; rel="last/)
```

Comment:
@kapadia I think you should use the rel="next" link header rather than constructing the URLs yourself: https://developer.github.com/v3/#pagination